File: src/main/java/dev/kikugie/techutils/config/modmenu/ModMenuIntegration.java
package dev.kikugie.techutils.config.modmenu;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import dev.kikugie.techutils.config.ConfigGui;

public class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return screen -> {
            ConfigGui gui = new ConfigGui();
            gui.setParent(screen);
            return gui;
        };
    }
}

File: src/main/kotlin/dev/kikugie/techutils/client/config/compat/ModMenuIntegration.kt
package dev.kikugie.techutils.client.config.compat

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import dev.kikugie.techutils.client.gui.config.TechUtilsConfigGui
import net.minecraft.client.gui.screen.Screen


class ModMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<Screen> =
        ConfigScreenFactory<Screen> { screen: Screen? -> TechUtilsConfigGui(screen) }
}